xc/xenops_server_xen.ml

```diff
@@ -3507,6 +3507,25 @@ let look_for_xenctrl () =
       exit 1;
     end
 
+let watch_xen_event () =
+  let open Xeneventchn in
+  let handle = init () in
+  let virq_enomem_port = bind_virq handle Enomem in
+  let host_uuid = Inventory.lookup Inventory._installation_uuid in
+  let rec process_event () =
+    debug "pending xen event...";
+    let port = pending handle in
+    if port = virq_enomem_port then begin
```

Comment:
I think it would be clearer to handle this with a `match` to avoid if-then-else chains.
```
let event = pending handle in
match event with
| port when port = virq_enomem_port ->
| port when port <> -1 ->
| otherwise -> debug "received xen event %d (%s)" event __LOC__; (* creates too much logging? *)
```

Reply:
Please note the logic under port when `port <> -1` should also be executed when `port = virq_enomem_port`, so the match is incorrect.
I don't use `match` here because we just have one type of port now.

Reply:
I see your point. I'm just a bit nervous about code with `if` without `else` and incomplete case handling.